### PR TITLE
Rcov tests (in progress)

### PR DESCRIPTION
**Pull request by warpc**

### CODE REVIEW

**Comment by warpc** [comment]
retest this please

**Comment by warpc** [comment]
retest this please

**Comment by warpc** [comment]
retest this please

**Comment by warpc** [comment]
retest this please

**Comment by warpc** [comment]
retest this please

**Comment by warpc** [comment]
retest this please

**Comment by warpc** [comment]
retest this please

**Comment by warpc** [comment]
retest this please

**Comment by warpc** [comment]
retest this please

**Comment by warpc** [comment]
retest this please

**Comment by warpc** [comment]
retest this please

File: .ruby-version [hunk 1]
```diff
@@ -0,0 +1 @@
+1.9.3
```
Comment (mihgen) [inline .ruby-version]: What is this for?

File: Rakefile [hunk 2]
```diff
@@ -0,0 +1,20 @@
+require 'rspec/core/rake_task'
+
+rspec_opts = "--color --format documentation"
+
+RSpec::Core::RakeTask.new(:spec, 'spec:unit') do |t|
```
Comment (mihgen) [inline Rakefile]: I'm not sure if this block is used anywhere

File: Rakefile [hunk 3]
```diff
@@ -0,0 +1,20 @@
+require 'rspec/core/rake_task'
+
+rspec_opts = "--color --format documentation"
```
Comment (mihgen) [inline Rakefile]: Looks like this is used in only one place

File: .ruby-version (hunk 1 above)
Comment (warpc) [inline .ruby-version]: Now it is official way to say what ruby should be used in production and development. Also this file support from box RVM and Rbenv (automatically select ruby version for project or if such version of ruby not installed, suggest to install it).

I think it is good way to say community about ruby version.

File: Gemfile [hunk 4]
```diff
@@ -0,0 +1,4 @@
+source 'https://rubygems.org'
+source 'http://download.mirantis.com/fuelweb-repo/3.2/gems/'
+
+gemspec
```
Comment (warpc) [inline Gemfile]: As .ruby-version it is good way to inform developers from community about sources of packages and give ability to quick start using only one default command: `bundle install` to start development.

But without mcollective-client gem in http://download.mirantis.com/fuelweb-repo/3.2/gems point about quick start not work

File: utils/jenkins/astute_rspec_check.sh [hunk 5]
```diff
@@ -0,0 +1,63 @@
...
+    # cd $WORKSPACE
+    # 
+    # # Clean up work folder
+    # rm -Rf mcollective_tmp *.zip
+    curl -LO https://dl.dropboxusercontent.com/u/984976/mcollective-client-2.3.1.gem
+    gem install mcollective-client-2.3.1.gem
+    rm mcollective-client-2.3.1.gem
+    echo 'source "https://rubygems.org"' > Gemfile
+    echo 'source "http://download.mirantis.com/fuelweb-repo/3.2/gems/"' >> Gemfile
+    echo 'gemspec' >> Gemfile
```
Comment (warpc) [inline utils/jenkins/astute_rspec_check.sh]: Here i expect what Gemfile file (https://github.com/Mirantis/astute/pull/66/files#L2R1) not exist. If we decide to save it and place mcollective-client to http://download.mirantis.com/fuelweb-repo/3.2/gems/ this block (curl --- gemspec) can be safely remove.

**Comment by warpc** [comment]
retest this please

**Comment by warpc** [comment]
retest this please

**Comment by warpc** [comment]
retest this please

**Comment by warpc** [comment]
Ready for final review and merge

**Comment by yoojinl** [comment]
++
Where can we see result of rcov? Can we see them somewhere in jenkins?

**Comment by warpc** [comment]
Developers can see information about it in <project folder>/coverage/rcov/index.html. After every Rspec run developer will get message like this:

```
Finished in 53.56 seconds
132 examples, 0 failures, 2 pending
Coverage report Rcov style generated for RSpec to <project folder>/coverage/rcov
```

Now i will setup Rcov in Jenkins

**Comment by warpc** [comment]
retest this please

**Pull request merged**